src/Aquarius.Client.UnitTests/Samples/Client/LazyGetTests.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            if (resultCount >= totalCount)
+                throw new ArgumentOutOfRangeException(nameof(resultCount), $"ResultCount={resultCount} must be strictly less than TotalCount={totalCount}");
+
+            return CreateResults(resultCount, totalCount);
+        }
+
+        private ThingResults CreateResultsWithNoCursor(int resultCount, int totalCount)
+        {
+            var results = CreatePartialResults(resultCount, totalCount);
+
+            results.Cursor = null;
+
+            return results;
+        }
+
+        private ThingResults CreateResultsWithEmptyCursor(int resultCount, int totalCount)
+        {
+            var results = CreatePartialResults(resultCount, totalCount);
+
+            results.Cursor = string.Empty;
+
+            return results;
+        }
+
+        private ThingResults CreateResults(int resultCount, int totalCount)
+        {
+            var results = new ThingResults
+            {
+                Cursor = _fixture.Create<string>(),
+                DomainObjects = new List<Thing>(),
+                TotalCount = totalCount
+            };
+
+            if (resultCount > 0)
+            {
+                results.DomainObjects = _fixture.CreateMany<Thing>(resultCount).ToList();
+            }
+            else
+            {
+                results.Cursor = null;
+            }
+
+            return results;
+        }
+
+        private List<Thing> EvaluateAllLazyLoadedItems()
+        {
+            return _client.LazyGet<Thing, GetThings, ThingResults>(new GetThings()).DomainObjects.ToList();
+        }
+
+        private void AssertExpectedItemsAndLazyFetches(List<Thing> items, int expectedTotalCount, int expectedFetchesCount)
+        {
+            items.Count.ShouldBeEquivalentTo(expectedTotalCount, "Total count of items should match");
+
+            _mockServiceClient
+                .Received(expectedFetchesCount)
+                .Get(Arg.Any<GetThings>());
+        }
+
+        [Test]
+        public void LazyGet_SecondResponseFails_Throws()
+        {
+            _mockServiceClient
+                .Get(Arg.Any<GetThings>())
+                .Returns(
+                    x => CreatePartialResults(1, 2),
+                    x => { throw new ArgumentException();});
+
+            ((Action) (() => EvaluateAllLazyLoadedItems())).ShouldThrow<ArgumentException>();
+        }
+
+        [Test]
+        public void LazyGet_SecondResponseContainsAllRemainingItems_Succeeds()
+        {
+            _mockServiceClient
+                .Get(Arg.Any<GetThings>())
+                .Returns(
+                    x => CreatePartialResults(1, 3),
+                    x => CreateFinalResults(1, 2));
+
+            var items = EvaluateAllLazyLoadedItems();
+
+            AssertExpectedItemsAndLazyFetches(items, 2, 2);
+        }
+
+        [Test]
+        public void LazyGet_SecondResponseWithNoCursor_Succeeds()
```

Comment:
This isn't actually a thing that samples does.

There is a case where the first result has no cursor. In that case, you don't want to call any more, because you'll keep getting the same result.